linuxNew/jdk/redhat/src/main/packaging/Dockerfile

```diff
@@ -32,37 +31,11 @@ RUN --mount=type=secret,id=gpg,gid=10001,uid=10000,dst=/tmp/private.gpg \
 '\
 >> /home/builder/.rpmmacros; \
 	fi
-USER root
 
 # Prepare entrypoint and build scripts
 ADD entrypoint.sh /entrypoint.sh
-RUN printf '#!/usr/bin/env bash\n\
-set -euxo pipefail\n\
-\n\
-# Ensure necessary directories for rpmbuild operation are present.\n\
-rpmdev-setuptree \n\
-\n\
-# Build all spec files we can fin.\n\
-targets="x86_64 ppc64le s390x aarch64 armv7hl" \n\
-for spec in "$(ls /home/builder/build/generated/packaging/*.spec)"; do\n\
-	spectool -g -R "$spec";\n\
-	rpmbuild --nodeps -bs "$spec"; \n\
-	if [[ "$spec" =~ "temurin-8-jdk" ]]; then \n\
-		targets="x86_64 ppc64le aarch64 armv7hl" \n\
-	fi \n\
-	for target in $targets; do \n\
-		rpmbuild --target "$target" --rebuild /home/builder/rpmbuild/SRPMS/*.src.rpm;\n\
-	done;\n\
-done;\n\
-\n\
-# Copy generated SRPMS, RPMs to destination folder\n\
-find /home/builder/rpmbuild/SRPMS /home/builder/rpmbuild/RPMS -type f -name "*.rpm" -print0 | xargs -0 -I {} cp {} /home/builder/out\n\
-# Sign generated RPMs with rpmsign.\n\
-if grep -q %%_gpg_name /home/builder/.rpmmacros; then \n\
-	rpmsign --addsign /home/builder/out/*.rpm \n\
-fi;\n\
-'\
->> /home/builder/build.sh
+ADD build.sh /home/builder/build.sh
+USER root
```

Comment:
Why has this been added explicitly?

Reply:
from https://github.com/adoptium/installer/blob/master/linuxNew/jdk/redhat/src/main/packaging/Dockerfile#L35
we use "root" to perform entrypoint.
so this is not added in the change.

and  this change, in order to run "chmod +x" for build.sh
otherwise, permission denied. 
see log https://github.com/adoptium/installer/runs/6656938909?check_suite_focus=true

` 
RUN chmod +x /home/builder/build.sh:  
executor failed running [/bin/sh -c chmod +x /home/builder/build.sh]: exit code: 1`

Reply:
Do you know how it worked before the change? The `chmod` isn't new in here.

Reply:
While I'm still not sure why the `USER root` statement was moved it doesn't negatively impact the functionality so I'm ok with it now.